tests/mellowLpRouter.test.ts

```diff
@@ -55,62 +55,92 @@ describe('Mellow Router Test Suite', () => {
+        erc20RootVaultAddresses[0],
+      );
+
+      const erc20RootVaultGovernanceContract = new ethers.Contract(
+        erc20RootVaultAddresses[0],
+        Erc20RootVaultABI,
+        provider,
+      );
+      const erc20RootVaultGovernanceAddress =
+        await erc20RootVaultGovernanceContract.vaultGovernance();
+      expect(ethMellowLpRouter.readOnlyContracts?.erc20RootVaultGovernance[0].address).to.be.eq(
+        erc20RootVaultGovernanceAddress,
+      );
+
+      expect(ethMellowLpRouter.readOnlyContracts?.mellowRouterContract.address).to.be.eq(
+        mellowRouterContract.address,
+      );
     });
 
     it('Check that user has been initialised correctly', async () => {
-      const mellowRouter = ethMellowLpRouter;
-
       // Get the router contract
       const mellowRouterContract = new ethers.Contract(
         ethMellowRouterAddress,
         MellowMultiVaultRouterABI,
         provider,
       );
 
-      expect(mellowRouter.writeContracts?.mellowRouter.address).to.be.eq(
+      expect(ethMellowLpRouter.writeContracts?.mellowRouter.address).to.be.eq(
         mellowRouterContract.address,
       );
-    });
-
-    it('User deposits eth into router contract with uneven split', async () => {
-      const mellowRouter = ethMellowLpRouter;
-      const weights = [0.2, 0.8]; // Need to know how many different maturities there are to know the split
+      const erc20RootVaultAddresses = await mellowRouterContract.getVaults();
+      expect(ethMellowLpRouter.writeContracts?.erc20RootVault[0].address).to.be.eq(
+        erc20RootVaultAddresses[0],
+      );
 
-      await mellowRouter.deposit(depositAmount, weights);
+      const tokenAddress = await mellowRouterContract.token();
+      expect(ethMellowLpRouter.readOnlyContracts?.token.address).to.be.eq(tokenAddress);
+    });
 
-      const userWalletBalanceBefore = await userWallet.getBalance();
-      const userWalletBalanceAfter = userWalletBalanceBefore.toNumber() - 1;
+    it('User deposits eth into router contract with even split', async () => {
+      const weights = [50, 50]; // Needs to sum to 100
 
-      expect(userWalletBalanceBefore).to.be.eq(userWalletBalanceAfter);
+      const receipt = await ethMellowLpRouter.deposit(depositAmount, weights);
+      // eslint-disable-next-line
+      console.log("tx hash: ", receipt.transactionHash);
+      // fails at the moment because goerli caps are not set
+      expect(ethMellowLpRouter.vaultCap).to.be.greaterThan(ethMellowLpRouter.vaultCumulative);
     });
-    it('User deposits eth into router contract 1:0 split', async () => {
-      const mellowRouter = ethMellowLpRouter;
-      const weights = [1, 0];
-
-      await mellowRouter.deposit(depositAmount, weights);
+    it('User deposits eth into router contract uneven split', async () => {
+      const weights = [100, 0];
+      await ethMellowLpRouter.deposit(depositAmount, weights);
```

Comment:
these tests are weird... you create action but no assertion... can you explain this pattern?

Reply:
Yes, they were placeholders until I got the scenarios right. This PR was a WIP. I have submitted the set for review now.